tests/integration/conftest.py

```diff
@@ -74,3 +76,28 @@ def oci_image(metadata_file: str, image_name: str) -> str:
             raise ValueError("Upstream source not found")
 
         return upstream_source
+
+    async def juju_run(self, unit, cmd):
+        """Run a command on a unit and return the output."""
+        action = await unit.run(cmd)
+        result = await action.wait()
+        code = result.results["return-code"]
+        stdout = result.results.get("stdout")
+        stderr = result.results.get("stderr")
+        assert code == 0, f"{cmd} failed ({code}): {stderr or stdout}"
+        return stdout
+
+    async def query_prometheus(self, ops_test, query):
+        prometheus_unit = ops_test.model.applications["prometheus-k8s"].units[0]
+        assert prometheus_unit.workload_status == "active"
+
+        # Query pods.
+        query = {"query": query}
+        qs = urlencode(query)
+        url = f"http://localhost:9090/api/v1/query?{qs}"
+        output = await self.juju_run(prometheus_unit, f"curl '{url}'")
```

Comment:
I would prefer making the request outside the unit to make sure the service and port is properly exposed